packages/ckb-sdk-core/examples/sendTransaction.js

```diff
@@ -0,0 +1,281 @@
+    const targetCapacity = BigInt(capacity)
+
+    /**
+     * the new cell for next owner
+     */
+    const targetOutput = {
+      capacity: targetCapacity,
+      lock: {
+        codeHash: SYSTEM_ENCRYPTION_CODE_HASH,
+        args: [targetBlake160edPublicKey],
+      },
+      data: '0x',
+    }
+
+    /**
+     * the new cell as a change
+     */
+    const chargeOutput = {
+      capacity: 0n,
+      lock: {
+        codeHash: SYSTEM_ENCRYPTION_CODE_HASH,
+        args: [`0x${blake160edPublicKey}`],
+      },
+      data: '0x',
+    }
+
+    // load the unspent cells in Promise method, just an optimiztion of code.
+    const loadCells = () =>
+      new Promise((resolve, reject) => {
+        core.rpc
+          .getTipBlockNumber()
+          .then(tipNumber =>
+            getUnspentCells(lockHash, 0, tipNumber, cells => {
+              resolve(cells)
+            })
+          )
+          .catch(reject)
+      })
+
+    const unspentCells = await loadCells()
+    const inputs = []
+    let inputCapacity = 0n
+    /**
+     * pick inputs
+     */
+    for (let i = 0; i < unspentCells.length; i++) {
+      const unspentCell = unspentCells[i]
+      inputs.push({
+        previousOutput: unspentCell.outPoint,
+        since: '0',
+        args: [],
+      })
+      inputCapacity += BigInt(unspentCells[1].capacity)
+      if (inputCapacity >= targetCapacity) {
+        break
+      }
+    }
+    if (inputCapacity < targetCapacity) {
+      throw new Error('inputCapacity is not enough')
+    }
+    if (inputCapacity > targetCapacity) {
+      chargeOutput.capacity = inputCapacity - targetCapacity
+    }
+
+    /**
+     * compose the raw transaction which has an empty witnesses
+     */
+    const tx = {
+      version: '0',
+      deps: [SYSTEM_ENCRYPTION_OUT_POINT],
+      inputs,
+      outputs:
+        chargeOutput.capacity > 0n ? [
+          {
+            ...targetOutput,
+            capacity: targetOutput.capacity.toString(),
+          },
+          {
+            ...chargeOutput,
+            capacity: chargeOutput.capacity.toString(),
+          },
+        ] : [
+          {
+            ...targetOutput,
+            capacity: targetOutput.capacity.toString(),
+          },
+        ],
+      witnesses: [
+        {
+          data: [],
+        },
+      ],
+    }
+    return tx
+  }
+
+  /**
+   * to see the generated transaction
+   */
+  // generateTransaction(`0x${blake160edPublicKey}`, 1000000000000).then(tx => {
+  //   console.log(JSON.stringify(tx, null, 2))
+  // })
+
+  /**
+   * add an RPC which calculates the transaction hash, it can also be calculated offline
+   * @ref: https://github.com/nervosnetwork/ckb/blob/develop/rpc/json/rpc.json
+   * @description this RPC is marked as deprecated, but still works for now.
+   *              this RPC sends the raw transaction to CKB and returns the hash of the transaction.
+   */
+  const computeTransactionHashMethod = {
```

Comment:
Should we provide this method in `RPC` ? @ashchan

Reply:
It's a must ATM.